src/lib/stores/accountsStore.ts

```diff
@@ -41,17 +43,25 @@ export async function fetchAccounts(
     const allAccounts = await thisWeb3Accounts();
     allAccounts.forEach((a) => {
       // include only the accounts allowed for this chain
-      if (!a.meta.genesisHash || selectedNetworkInfo.genesisHash === a.meta.genesisHash) {
+      if (!a.meta.genesisHash || selectedNetwork.genesisHash === a.meta.genesisHash) {
         foundAccounts[a.address] = a;
       }
     });
   }
 
-  const foundProviderAccounts: AccountMap | MetaMap = {};
-  for (const index in Object.keys(foundAccounts)) {
-    const account = Object.values(foundAccounts)[index];
-    const { isProvider } = await getMsaInfo(apiPromise, account.address);
-    if (isProvider) foundProviderAccounts[account.address] = account;
+  // Segment into provider accounts and non-provider accounts
+  const regularAccounts: string[] = [];
+  const providerAccounts: string[] = [];
+
+  for (const address in foundAccounts) {
+    const { isProvider } = await getMsaInfo(apiPromise, address);
+    if (isProvider) {
+      providerAccounts.push(address);
+    } else {
+      regularAccounts.push(address);
+    }
   }
-  storeProviderAccounts.set(foundProviderAccounts);
+
+  storeProviderAccounts.set(providerAccounts);
+  storeValidAccounts.set(regularAccounts);
```

Comment:
I thought that valid accounts were all accounts (provider or not).

Additionally, what is your thought on storing non-provider accounts? When will we be using those?

In my story (#88 ) I wrote some code to check if an address is not associated with an MSA because in that case, it would be an eligible key to add to an MSA Id in our Add Key modal. Was that the case you were thinking of?

Reply:
That was from the existing model so I didn't remove it.  If we end up not using it, we can remove it but it sounds like you have a use case.